variables.tf

```diff
@@ -29,33 +29,31 @@ variable "policy" {
 
 variable "policy_attachment_name" {
   type        = string
-  description = "(Required) - The name of the attachment. This cannot be an empty string."
+  description = "(Optional) - The name of the attachment. Required when users, roles or groups are provided."
+  default     = null
 }
 
 variable "users" {
   type        = list(string)
   description = "(Optional) - The user(s) the policy should be applied to"
-  default     = []
+  default     = null
 }
 
 variable "roles" {
   type        = list(string)
   description = "(Optional) - The role(s) the policy should be applied to"
-  default     = []
+  default     = null
 }
 
 variable "groups" {
   type        = list(string)
   description = "(Optional) - The group(s) the policy should be applied to"
-  default     = []
+  default     = null
 }
```

Comment:
You may need to maintain the default values as `[ ]` . See comment on `locals.tf` and `main.tf`